src/SDK/SmartDetectorsSDK/AlertPresentation/AzureResourceManagerRequest.cs

```diff
@@ -0,0 +1,123 @@
+            }
+
+            if (requestUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("The ARM request URI must be a relative URI", nameof(requestUri));
+            }
+
+            this.RequestUri = requestUri;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureResourceManagerRequest"/> class, based on the given resource and suffix.
+        /// </summary>
+        /// <param name="resource">The resource for which the ARM request should be made</param>
+        /// <param name="requestSuffix">An optional suffix for the request, which will appended to the resource's URI</param>
+        protected AzureResourceManagerRequest(ResourceIdentifier resource, string requestSuffix)
+        {
+            if (resource == null)
+            {
+                throw new ArgumentNullException(nameof(resource));
+            }
+
+            // Make sure that we'll be able to append the request suffix
+            if (string.IsNullOrEmpty(requestSuffix))
+            {
+                this.RequestUri = new Uri(resource.ToResourceId(), UriKind.Relative);
+            }
+            else if (requestSuffix[0] != '/' && requestSuffix[0] == '?')
+            {
+                this.RequestUri = new Uri($"{resource.ToResourceId()}/{requestSuffix}", UriKind.Relative);
+            }
+            else
+            {
+                this.RequestUri = new Uri($"{resource.ToResourceId()}{requestSuffix}", UriKind.Relative);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureResourceManagerRequest"/> class, based on an Azure Monitor log
+        /// query.
+        /// </summary>
+        /// <param name="resource">The resource for which to make the query</param>
+        /// <param name="query">The query to run</param>
+        /// <param name="queryTimeSpan">Optional time span to use for limiting the query data.</param>
+        protected AzureResourceManagerRequest(ResourceIdentifier resource, string query, TimeSpan? queryTimeSpan)
+        {
+            if (resource == null)
+            {
+                throw new ArgumentNullException(nameof(resource));
+            }
+
+            if (string.IsNullOrEmpty(query))
+            {
+                throw new ArgumentNullException(nameof(query));
+            }
+
+            // Compose the request URL (different handling for Application Insights vs. Log Analytics)
+            string apiVersion;
+            string queryUrlPath;
+            if (resource.ResourceType == ResourceType.ApplicationInsights)
+            {
+                // Applications Insights query API format is: {resource}/api/query?query={query}&timespan={queryTimeSpan}&api-version=2018-04-20
+                apiVersion = "2018-04-20";
+                queryUrlPath = $"{resource.ToResourceId()}/api/query";
+            }
+            else
```

Comment:
>else [](start = 12, length = 4)

I suggest to verify that the ResourceType is LogAnalytics here, and throw an exception if it is not

Reply:
actually not - this is the API for resource-centric queries

naturally, i have no way of testing it right now, but it should work

---
In reply to: [279190755](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/61#discussion_r279190755) [](ancestors = 279190755)